### DISCUSSION TITLE
Li-ion Battery Temperature change

### DISCUSSION THREAD

**Discussion by Alikarbar** [community/community#128847] [Programming Help]
### Select Topic Area

Question

### Body

Hi every one.
I am new in PyBamm, I need help to know how I can plot temperature cell during a single charge or discharge time , I mean I need a graph of Cell Temp VS Time. I would appreciate it if some one can lead me.

**Comment by buu-huu** [comment]
Can you provide source code? That would make it easier to help in your specific context :)

But anyways, here is a short example to get you started:

```python
import pybamm
import matplotlib.pyplot as plt

# Step 1: Choose and create a battery model
model = pybamm.lithium_ion.DFN()  # Doyle-Fuller-Newman model

# Step 2: Define the parameter values and create a simulation object
param = model.default_parameter_values
experiment = pybamm.Experiment(["Charge at 1C until 4.2V", "Rest for 1 hour", "Discharge at 1C until 2.5V"])

# Step 3: Create a simulation
sim = pybamm.Simulation(model, experiment=experiment, parameter_values=param)

# Step 4: Run the simulation
sim.solve()

# Step 5: Extract the temperature data and time
t = sim.solution["Time [s]"].entries
T = sim.solution["X-averaged cell temperature [K]"].entries

# Step 6: Plot the temperature vs time
plt.figure(figsize=(10, 5))
plt.plot(t, T, label='Cell Temperature')
plt.xlabel('Time [s]')
plt.ylabel('Temperature [K]')
plt.title('Cell Temperature vs Time')
plt.legend()
plt.grid(True)
plt.show()

```

**Reply by ebndev** [reply]
Hi @buu-huu, thanks for taking the time to answer!

@Alikarbar if the response above helped, please feel encouraged to mark it as the answer. If not, let us know if you have further questions or post the solution if you've resolved it on your own. It would be helpful for other users who might experience this in the future 😄

**Reply by Alikarbar** [reply]
Thanks for your reply, I'm actually trying to simulate a li-ion battery based on new parameters achieved latest, using DFN coupled with thermal model. I'm going to investigate in Cell Temperature [C] as the variable and see how a single charging from 0 to 100 soc affects  cell temp, the variable  you mentioned works, but cell temp doesn't work in this way.
